openconcept/aerodynamics/openaerostruct/tests/test_wave_drag.py

```diff
@@ -0,0 +1,102 @@
+"""
+@File    :   test_wave_drag.py
+@Date    :   2023/04/17
+@Author  :   Eytan Adler
+@Description : Test the wave drag utilities
+"""
+
+# ==============================================================================
+# Standard Python modules
+# ==============================================================================
+import unittest
+
+# ==============================================================================
+# External Python modules
+# ==============================================================================
+import openmdao.api as om
+from openmdao.utils.assert_utils import assert_near_equal, assert_check_partials
+
+# ==============================================================================
+# Extension modules
+# ==============================================================================
+from openconcept.aerodynamics.openaerostruct import WaveDragFromSections
+
+
+class WaveDragFromSectionsTestCase(unittest.TestCase):
+    def test_verify_OAS(self):
+        """
+        Verify against OpenAeroStruct
+        """
+        p = om.Problem()
+        p.model.add_subsystem("comp", WaveDragFromSections(num_sections=4), promotes=["*"])
+        p.setup()
+
+        p.set_val("fltcond|M", 0.9)
+        p.set_val("fltcond|CL", 0.29657041)
+
+        p.set_val("toverc_sec", [0.08, 0.11, 0.15, 0.15])
+        p.set_val("y_sec", [-105, -41, -23], units="ft")
+        p.set_val("chord_sec", [8, 25, 75, 104], units="ft")
+        p.set_val("c4sweep", 37.69966129860839, units="deg")
+
+        p.run_model()
+
+        assert_near_equal(p.get_val("CD_wave"), 0.001520816219023791, tolerance=1e-3)
```

Comment:
Just added this. The values slightly changed because I'm now doing a weighted average of cos(section sweep) rather than of section sweep directly. This makes only a small difference but is now consistent with OpenAeroStruct and more correct.